### ISSUE TITLE
Release Gutenberg 15.1

### ISSUE THREAD

**Issue by juanmaguitar** [WordPress/gutenberg#47411]
This issue is to provide visibility on the progress of the release process of Gutenberg 15.1 and to centralize any conversations about it.
The ultimate goal of this issue is to keep the reference of the steps, resources, work, and conversations about this release so it can be helpful for the next contributors releasing a new Gutenberg version.

---

- Gutenberg version to release: **15.1** ([milestone](https://github.com/WordPress/gutenberg/milestone/196))
- Release Manager (a.k.a. Release Lead): @juanmaguitar 
- Release Date 15.1 RC: 1st Feb _(it was supposed to be on Jan 25th, but [it's been delayed to 1st Feb](https://wordpress.slack.com/archives/C04JZ199XPG/p1674554615188169))_
- Release Date 15.1: 8th Feb
- Release Weeks: Weeks 4, 5 & 6 (from 23rd Jan to 12th Feb) _[three weeks period instead of two, due to this delay](https://wordpress.slack.com/archives/C04JZ199XPG/p1674554615188169)_
- Release Manager Buddy: @mburridge _(led the [release of GB 15.0](https://make.wordpress.org/core/2023/01/20/whats-new-in-gutenberg-15-0-18-january/))_
- Previous version changelog _(as a reference)_: [15.0](https://github.com/WordPress/gutenberg/releases/tag/v15.0.0)

Resources:
- [Leading a Gutenberg Plugin Release](https://codep2.wordpress.com/2021/07/22/leading-a-gutenberg-plugin-release/)
- [Gutenberg Release Process | Block Editor Handbook | WordPress Developer Resources](https://developer.wordpress.org/block-editor/contributors/code/release/)

## Checklist

> **Note**
> A new major version of Gutenberg is released approximately every two weeks

### Preparation (before Wednesday Feb1st)
_preparation usually happens before Wednesday Week 1_

- [ ] 📖 _Read:_ [Gutenberg Release Process](https://developer.wordpress.org/block-editor/contributors/code/release/)
- [ ] 📽 _Watch:_ [Gutenberg Plugin: New Release Workflow](https://www.youtube.com/watch?v=TnSgJd3zpJY)
- [ ] 📽 _Watch:_ [Creating the Gutenberg plugin v12.0 Release Candidate](https://www.youtube.com/watch?v=FLkLHKecxWg)
- [ ] 📽 _Watch:_ [Gutenberg plugin v12.0.0 Release Party!](https://www.youtube.com/watch?v=4SDtpVPDsLc)

###  RC Day (Wednesday Feb 1st)
_RC Day usually is on Wednesday Week 1_

- [ ] _Optional:_ Attend `#core-editor` meeting (Weekly meetings held Wednesdays at 14:00 UTC at the Slack channel)
- [x] Post a [message](https://wordpress.slack.com/archives/C02QB2JS7/p1674637170352029) in `#core-editor` channel to let folks know you are leading the next GB release 
- [x] Post a [message](https://wordpress.slack.com/archives/C02QB2JS7/p1674637170352029) in `#core-editor` channel asking for a "Release Manager Buddy" to support you with the release _(last person who did the release is a good candidate to help)_
- [x]  Post a [message](https://wordpress.slack.com/archives/C02QB2JS7/p1675155705836649) in `#core-editor` channel to let folks know you are starting the RC release process 
- [x] Organize and Label PRs on the relevant milestone
- [x] Create Draft Release
- [x] Changelog Created and Organized on Draft Release
- [x] Publish (RC) Release
- [x] [Announce](https://wordpress.slack.com/archives/C02QB2JS7/p1675253355327909) in `#core-editor` channel  that RC1 has been released and is ready for testing
- [x] Ping any other relevant channels announcing that the RC is available
- [x] Create [Draft](https://docs.google.com/document/d/1ejfJj2zgKCc-6P61FOjSm_2BduUZaMutNz4hmbCODdE/edit) of Release post on Make Core blog _([example](https://docs.google.com/document/d/1quftMN3zhKmph_OTGe4ZiA4pi0ftaA_nBbAU52HG9Dk/edit#))_

### Between RC and Release (from Wed Feb 1st to Wed Feb 8th)
_This period usually happens from Wednesday Week 1 to Wednesday Week 2_
- [x] [Backport to RC](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22Backport+to+Gutenberg+RC%22+is%3Aclosed)
- [x] [Draft Release Post Highlights and changelog](#)
- [x] Get assets from [Design team](https://make.wordpress.org/design/) for the post
- [x] Reach out to Highlight Authors to draft sections _(if necessary)_

### Release Day (Wednesday 8th February)
_Release Day usually is on Wednesday Week 2_

- [x] Ensure there are not pending [Backport to RC](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22Backport+to+Gutenberg+RC%22+is%3Aclosed)
  - Is this done automatically when running the `.zip` workflow?
  - Is this operation only needed for minor releases?
- [x] Check there are [no open PR's in the proper milestone](https://github.com/WordPress/gutenberg/pulls?q=is%3Aopen+is%3Apr+milestone%3A%22Gutenberg+15.1%22)
- [x] Check the release branch (`release/15.1`)  build has no visible regressions
- [x] Post [a message](https://wordpress.slack.com/archives/C02QB2JS7/p1675848099953319) in `#core-editor` channel to let folks know you are starting the release process
- [x] Run [Build Gutenberg Plugin Zip Action](https://github.com/WordPress/gutenberg/actions/workflows/build-plugin-zip.yml) workflow for the `stable` version
- [x] Organize and Label PRs on the relevant milestone
- [x] Create Draft Release
- [x] Changelog Created and Organized on Draft Release
- [x] Publish Release
- [x] Trigger the update to the plugin directory - SVN _(get approval from member of Core team if necessary)_
- [x] Announce in `#core-editor` channel that the plugin has been released
- [ ] Reach out to other contributors to help get the post reviewed
- [ ] Publish Release post on Make Core blog

_Thanks to @DaisyOlsen who originally created this checklist_

**Comment by juanmaguitar** [comment]
## Organize and Label PRs on the relevant milestone

To simplify the final organization of the Changelog (generated automatically from the labels of the PRs), in this step we should ensure that every PR on the [relevant milestone](https://github.com/WordPress/gutenberg/milestone/196) has at least one of the following labels:

```
'[Feature] Navigation Screen': 'Experiments',
'[Package] Dependency Extraction Webpack Plugin': 'Tools',
'[Package] Jest Puppeteer aXe': 'Tools',
'[Package] E2E Tests': 'Tools',
'[Package] E2E Test Utils': 'Tools',
'[Package] Env': 'Tools',
'[Package] ESLint plugin': 'Tools',
'[Package] stylelint config': 'Tools',
'[Package] Project management automation': 'Tools',
'[Type] Project Management': 'Tools',
'[Package] Scripts': 'Tools',
'[Type] Build Tooling': 'Tools',
'Automated Testing': 'Tools',
'[Type] Experimental': 'Experiments',
'[Type] Bug': 'Bug Fixes',
'[Type] Regression': 'Bug Fixes',
'[Type] Feature': 'Features',
'[Type] Enhancement': 'Enhancements',
'[Type] New API': 'New APIs',
'[Type] Performance': 'Performance',
'[Type] Developer Documentation': 'Documentation',
'[Type] Code Quality': 'Code Quality',
'[Type] Security': 'Security',
```

_Note: The changelog generation code is defined at 
https://github.com/WordPress/gutenberg/blob/trunk/bin/plugin/commands/changelog.js#L64_

_Tip: Ensuring `[Type] ...` labels is usually enough_

Any PR without one of these labels will be placed under a generic "Various" section, so by ensuring these labels, we ensure the generated Changelog groups properly the PRs in relevant H1s and [we can avoid to](https://developer.wordpress.org/block-editor/contributors/code/release/#1-curating-the-changelog)...
> Move all entries under the Various section to a more appropriate section.

...once the Changelog is generated

As per [this](https://developer.wordpress.org/block-editor/contributors/code/release/#1-curating-the-changelog) we can ignore the `Mobile App - ...` PRs as most of them should be removed from the final Changelog
>Remove all mobile app pull request entries (the only exception to this rule is if the mobile PR also updates functionality for the web).

---

Some strategies to review/add labels:
1. Start with [those with no labels](https://github.com/WordPress/gutenberg/issues?q=is%3Aclosed+milestone%3A%22Gutenberg+15.1%22+no%3Alabel)
2. Review [those without relevant `[Type] ...` labels](https://github.com/WordPress/gutenberg/issues?q=is%3Aclosed+milestone%3A%22Gutenberg+15.1%22+-label%3A%22%5BType%5D+Experimental%22+-label%3A%22%5BType%5D+Project+Management%22+-label%3A%22%5BType%5D+Build+Tooling%22+-label%3A%22%5BType%5D+Bug%22+-label%3A%22%5BType%5D+Regression%22+-label%3A%22%5BType%5D+Feature%22+-label%3A%22%5BType%5D+Enhancement%22+-label%3A%22%5BType%5D+New+API%22++-label%3A%22%5BType%5D+Performance%22+-label%3A%22%5BType%5D+Developer+Documentation%22+-label%3A%22%5BType%5D+Code+Quality%22+-label%3A%22%5BType%5D+Security%22+)

**Comment by juanmaguitar** [comment]
## Create Draft Release

From the [**Build Gutenberg Plugin Zip** Action](https://github.com/WordPress/gutenberg/actions/workflows/build-plugin-zip.yml) we run the _Workflow_ . Ensure that the branch is set to "trunk", and type "rc" into the text field.

<img width="388" alt="Screenshot 2023-02-01 at 11 15 13" src="https://user-images.githubusercontent.com/422576/216015138-ccc89abf-8223-4964-859a-72adc15dda80.png">

This will generate a `15.1.0 RC1 (Draft)`. Once this "draft" is generated, the next step is to edit the Changelog, to reorganize to improve its readability and save the changes.
By saving the changes, we complete the release process by making it public.

**Comment by juanmaguitar** [comment]
## Changelog Created and Organized on Draft Release

Copy the generated Changelog and curate it in your favorite IDE

**Comment by juanmaguitar** [comment]
## Publish (RC) Release

Ensure the "Set as pre-release" is check when you publish the RC version by clicking the "Publish release" (green button)

<img width="1180" alt="Screenshot 2023-02-01 at 11 36 53" src="https://user-images.githubusercontent.com/422576/216020007-0b06ba0c-d7e0-4a12-8c65-613d906c6de0.png">

**Comment by juanmaguitar** [comment]
The draft of the Release post on Make Core blog is here:
https://docs.google.com/document/d/1ejfJj2zgKCc-6P61FOjSm_2BduUZaMutNz4hmbCODdE/edit#heading=h.z2d6gcgxx6yr

**Comment by juanmaguitar** [comment]
## Backport to RC

PRs labelled as "Backport to Gutenberg RC" have to be "cherry-picked" into the release/15.1  branch before releasing GB v15.1 (stable)

```
node -v → v14.21.2
npm -v → 6.14.17
```

To do that I've done

```
git checkout release/15.1
npm run other:cherry-pick "Backport to Gutenberg RC"
```

```
⬢  gutenberg  release/15.1 ⦾ npm run other:cherry-pick "Backport to Gutenberg RC"

> gutenberg@15.1.0-rc.1 other:cherry-pick /Users/juanmanuelgarrido/PROJECTS/2023/WORDPRESS/GUTENBERG/gutenberg
> node ./bin/cherry-pick.mjs "Backport to Gutenberg RC"

You are on branch "release/15.1".
This script will:
• Cherry-pick the merged PRs labeled as "Backport to Gutenberg RC" to this branch
• Ask whether you want to push this branch
• Comment on each PR
• Remove the label from each PR
The last two actions will be performed USING YOUR GITHUB ACCOUNT that
you've linked to your GitHub CLI (gh command)

Do you want to proceed? (Y/n)Y
$ git pull origin release/15.1 --rebase...
$ git fetch origin trunk...
Found the following PRs to cherry-pick:
   #47851 – Site editor: specify active state color for template navigation button
Fetching commit IDs...
Done!
   #47851 – 16d787686bd874819263 – Site editor: specify active st...
Trying to cherry-pick one by one...
Cherry-picking round 1:
   ✅  cherry-pick commit: dbc7928225  for PR: #47851 – Site editor: specify active st...
Cherry-picking finished!
Summary:
   ✅  1 PRs got cherry-picked cleanly
   ✅  0 PRs failed

About to push to origin/release/15.1
Do you want to proceed? (Y/n)Y
Commenting and removing labels...
✅ 47851: I just cherry-picked this PR to the release/15.1 branch to get it included in the next release: dbc7928225
Done!
```

This script takes care of everything:
-  Cherry-pick the merged PRs labeled as "Backport to Gutenberg RC" to this branch
-  Ask whether you want to push this branch
-  Comment on each PR
-  Remove the label from each PR

**References:**
- [Gutenberg Release Process  > Creating Release Candidate Patches (done via git cherry-pick)](https://developer.wordpress.org/block-editor/contributors/code/release/#creating-release-candidate-patches-done-via-git-cherry-pick)
- [Cherry-picking automation > How can I use it for a Gutenberg plugin release?](https://developer.wordpress.org/block-editor/contributors/code/release/auto-cherry-picking/#how-can-i-use-it-for-a-gutenberg-plugin-release)
